megamek/src/megamek/common/logging/Log4j.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * MegaMek - Copyright (C) 2018 - The MegaMek Team
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ */
+package megamek.common.logging;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.LoggerContext;
```

Comment:
While I was looking at ways to make this easier to use I noticed that you're using internal code that is not part of the public API.

Reply:
AFAIK `log4j-api` does not provide any mean for programmatic configuration (and certainly does not provide the `InputStream`/`OutputStream` implementations used in `Log4j.snatchSytemStreams()`).

As far as my understanding goes, `log4j-core` is the logging backend and `log4j-api` is a facade that one could use with different backends (doesn't make much sense for an app, but it allows a library that internally uses `log4j` to be integrated in an app that uses `logback` as a logging backend).

The `log4j-core` API will certainly be less stable than `log4j-api`, but it is documented and –IMHO– far from deserving the label of "internal code" :-) Specifically, the use of `org.apache.logging.log4j.core.LoggerContext` (complete with the horrible cast from `(LoggerContext) LogManager.getContext(false)`) is documented at https://logging.apache.org/log4j/2.x/manual/customconfig.html#AddingToCurrent

Reply:
BTW the dependency on `org.apache.logging.log4j.core` is basically the reason why I made this `Log4j` utility class: the idea was to concentrate all log4j config functionality in one place.

Outside `MegaMek` this stuff will only be used by `Princess` to set its own logging level.